site/app/controllers/MiscController.php

```diff
@@ -91,7 +91,7 @@ private function checkValidAccess($is_zip, &$error_string, $download_with_any_ro
         // END HACK
 
 
-        $possible_directories = array("config_upload", "uploads", "submissions", "results", "checkout", "forum_attachments", "uploads/course_materials");
+        $possible_directories = array("config_upload", "uploads", "submissions", "results", "checkout", "forum_attachments", "course_materials");
```

Comment:
Removing the 'uploads' prefix here causes the issues with `checkValidAccess`.  Because of a hack above `//END HACK`, all graders automatically pass this check, so only students would get this far anyway, and they shouldn't have anything other than read access.  Since this function will be killed off, its not a big deal, just worth noting.

Reply:
Removing the `uploads` here is along with removing it from all the other scripts too. Best to not reveal too much about the directory structure (todo eventually: relative paths)